### DISCUSSION TITLE
How to add reverse parsing? The mail server needs this feature

### DISCUSSION THREAD

**Discussion by ghost** [AdguardTeam/AdGuardHome#2977] [Q&A]
How to add reverse parsing? The mail server needs this feature

**Comment by ameshkov** [comment]
Please be more specific, it's unclear what exactly you want.

**Reply by ghost** [reply]
dns configuration reverse parsing, IP-to-host resolution。

![image](https://user-images.githubusercontent.com/51149477/115205192-e2764700-a12b-11eb-88ba-5381b5956926.png)
![image](https://user-images.githubusercontent.com/51149477/115205206-e5713780-a12b-11eb-95cd-8364061ebdcf.png)

**Reply by ameshkov** [reply]
You probably need to use `$dnsrewrite` filtering rules for this:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

**Reply by DavidOsipov** [reply]
Guys, as I see, @da-gongren has his/her own mail server. The trick with mail servers is that they do a reverse DNS lookups (SMTP Reverse DNS lookup) to check if another SMTP server has the correct hostname (PTR DNS entry). [Smt like that ](https://www.mailenable.com/kb/content/article.asp?ID=ME020206)

So here @da-gongren is wondering if Adguard Home has functionality to run a Reverse DNS lookup of an IP. I personally don't need this functionality, but it can be quite important for mail servers' admins,  @ameshkov

**Reply by ameshkov** [reply]
@DavidOsipov I understand, and `$dnsrewrite` rules do allow adding PTR records:

> `||4.3.2.1.in-addr.arpa^$dnsrewrite=NOERROR;PTR;example.net.` adds a PTR record for reverse DNS. Reverse DNS requests for `1.2.3.4` to the AdGuardHome DNS server will result in `example.net`.

Of course it's not a **real** DNS record, it's just AGH emulating it.
